src/APIRequestWrapper.php

```diff
@@ -41,30 +42,35 @@ public function __construct(
         RequestFactoryInterface $requestFactory,
         ?string $baseUrl = null,
         ?string $resource = null
-    )
-    {
+    ) {
         // Our HTTP Client needs to be able to follow redirects.
         $this->httpClient = new RedirectClientDecorator($baseHttpClient);
         $this->baseUrl = $baseUrl ?? self::CONFIG_BASE;
         $this->requestFactory = $requestFactory;
         $this->resource = $resource ?? self::UFC_ENDPOINT;
         $this->queryParams = [
-            'apiKey' => $apiKey, ...$extraQueryParams
+            'apiKey' => $apiKey,
+            ...$extraQueryParams
         ];
     }
 
     /**
-     * @throws ClientExceptionInterface|HttpRequestException
+     * @throws HttpRequestException|InvalidApiKeyException
      */
     public function get(): string
     {
-        // Prepare the URL with query params
-        $resourceURI = $this->baseUrl . '/' . ltrim($this->resource, '/') . '?' . http_build_query($this->queryParams);
-
-        $request = $this->requestFactory->createRequest('GET', $resourceURI);
+        try {
+            // Prepare the URL with query params
+            $resourceURI = $this->baseUrl . '/' . ltrim($this->resource, '/') . '?' . http_build_query(
+                    $this->queryParams
+                );
 
-        $response = $this->httpClient->sendRequest($request);
+            $request = $this->requestFactory->createRequest('GET', $resourceURI);
 
+            $response = $this->httpClient->sendRequest($request);
+        } catch (ClientExceptionInterface $e) {
```

Comment:
Restrict the RequestClient exceptions to the class using it; bubble Eppo exceptions only